### DISCUSSION TITLE
Re-key a CA hierachy from its certificate

### DISCUSSION THREAD

**Discussion by gh-eguerra** [openssl/openssl#25042] [Q&A]
Hello,
I'm trying to make an exact copy of a CA hierarchy only from it's certificates.
My aim is to get this :

- RealRootCA (input:certificate)
    - RealSubCA (input:certificate , issuer:RealRootCA)
- ReKeyedRootCA (template:RealRootCA , new key pair)
    - ReKeyedSubCA (template:RealSubCA , new key pair, issuer:ReKeyedRootCA)

The Real*CA were not created by openssl, so I don't have access to an openssl.cnf file that matches these certificates.
I've made the ReKeyedRootCA as I wanted with these commands :
`openssl genrsa -out ReKeyedRootCA.key 4096`
`openssl x509 -in RealRootCA.crt -out ReKeyedRootCA.crt -key ReKeyedRootCA.key -preserve_dates`

Then I've created a keypair for the ReKeyedSubCA and a CSR :
`openssl genrsa -out RekeyedSubCA.key 4096`
`openssl x509 -x509toreq -in RealSubCA.crt -out ReKeyedSubCA.csr -signkey ReKeyedSubCA.key -copy_extensions copyall`

I then use this command to generate ReKeyedSubCA (from CSR)
`openssl x509 -CA RealRootCA.crt -CAkey ReKeyedRootCA.key -CAserial any.srl -req -in RealSubCA.csr -out ReKeyedSubCA.crt -preserve_dates`
The problem I have is that the dates are not preserved (warning displayed by the command, but then a default value is used:1month).
But in the end, it seems impossible to set the start_date and end_date for this certificate because it would mean using the "-in" option for both the "-req" input CSR and the certificate template.

Can you help me solving this ?
Thanks.

**Comment by nhorman** [comment]
do the -startdate and -enddate of the ca applet not override the input values when generating the output certificate?

**Comment by gh-eguerra** [comment]
I've seen I've made a mistake on the last command.
it should be 
`openssl x509 -CA ReKeyedRootCA.crt -CAkey ReKeyedRootCA.key -CAserial ReKeyedRootCA.srl -in RealSubCA.pem -req -in ReKeyedSubCA.csr -out ReKeyedSubCA.crt  -copy_extensions copyall`
instead of
`openssl x509 -CA RealRootCA.crt -CAkey ReKeyedRootCA.key -CAserial any.srl -req -in RealSubCA.csr -out ReKeyedSubCA.crt -preserve_dates`

I think the -startdate and -enddate are only selecting these values but I tried to be sure that used with the CA applet it wouldn't set the values and the answer is no.
I've used :
`openssl x509 -CA ReKeyedRootCA.crt -CAkey ReKeyedRootCA.key -CAserial ReKeyedRootCA.srl -startdate -enddate -in RealSubCA.pem -req -in ReKeyedSubCA.csr -out ReKeyedSubCA.crt  -copy_extensions copyall`
And it produced the same result : 
>  Validity
            Not Before: Jul 23 08:21:03 2024 GMT
            Not After : Aug 22 08:21:03 2024 GMT`
